posthog/test/test_feature_flags.py

```diff
@@ -1149,6 +1373,122 @@ def test_flag_with_multiple_variant_overrides(self, patch_decide):
         # decide not called because this can be evaluated locally
         self.assertEqual(patch_decide.call_count, 0)
 
+    @mock.patch("posthog.client.decide")
+    def test_boolean_feature_flag_payloads_local(self, patch_decide):
+        basic_flag = {
+            "id": 1,
+            "name": "Beta Feature",
+            "key": "person-flag",
+            "is_simple_flag": True,
+            "active": True,
+            "filters": {
+                "groups": [
+                    {
+                        "properties": [
+                            {
+                                "key": "region",
+                                "operator": "exact",
+                                "value": ["USA"],
+                                "type": "person",
+                            }
+                        ],
+                        "rollout_percentage": 100,
+                    }
+                ],
+                "payloads": {"true": 300},
+            },
+        }
+        self.client.feature_flags = [basic_flag]
+        self.client.feature_flags_by_key = {"person-flag": basic_flag}
+
+        self.assertEqual(
+            self.client.get_feature_flag_payload(
+                "person-flag", "some-distinct-id", person_properties={"region": "USA"}
+            ),
+            300,
+        )
+
+        self.assertEqual(
+            self.client.get_feature_flag_payload(
+                "person-flag", "some-distinct-id", match_value=True, person_properties={"region": "USA"}
+            ),
+            300,
+        )
+        self.assertEqual(patch_decide.call_count, 0)
+
+    @mock.patch("posthog.client.decide")
+    def test_boolean_feature_flag_payload_decide(self, patch_decide):
+        patch_decide.return_value = {"featureFlagPayloads": {"person-flag": 300}}
+        self.assertEqual(
+            self.client.get_feature_flag_payload(
+                "person-flag", "some-distinct-id", person_properties={"region": "USA"}
+            ),
+            300,
+        )
+
+        self.assertEqual(
+            self.client.get_feature_flag_payload(
+                "person-flag", "some-distinct-id", match_value=True, person_properties={"region": "USA"}
+            ),
+            300,
+        )
+        self.assertEqual(patch_decide.call_count, 2)
```

Comment:
how is this calling decide twice? It doesn't in my spec 🤔

Reply:
because there are no saved flags / local evaluation is off / no personal API key.

So, we don't know what match_value=True corresponds to, flag wise and need to call decide.

Maybe your test has local eval on? In which case it ought to be 0 calls.

Reply:
Another q: What about a test when decide returns multivariate payloads?

Reply:
That code path should be identical because a decide-based return for payloads reads from `featureFlagPayloads` key which should contains <flag-key>: payload mapping. There's nothing that distinguishes multivariate flag payloads here. The primary difference exists on local evaluation return because of how the payloads are defined